game-core/src/main/java/games/strategy/triplea/ui/AbstractForumPosterPanel.java

```diff
@@ -47,26 +48,13 @@ public void display(final GamePlayer gamePlayer) {
         });
   }
 
-  protected abstract boolean allowIncludeTerritorySummary();
-
-  protected abstract boolean allowIncludeProductionSummary();
-
-  protected abstract boolean allowDiceBattleDetails();
-
-  protected abstract boolean allowDiceStatistics();
-
-  protected abstract IAbstractForumPosterDelegate getForumPosterDelegate();
-
-  protected abstract boolean getHasPostedTurnSummary();
+  protected abstract boolean includeDetailsAndSummary();
 
   protected abstract boolean skipPosting();
 
   protected abstract String getTitle();
 
-  @Override
-  public abstract String toString();
```

Comment:
This is really odd code, I don't think there is an effect here to the override. Until the 'play' button is fixed, should probably wait on manual verification of the component before removing.

The 'toString' gets passed as a constraints object (ActionButtons.java):
```
 add(battlePanel, battlePanel.toString());
    add(this.movePanel, this.movePanel.toString());
    add(repairPanel, repairPanel.toString());
    add(purchasePanel, purchasePanel.toString());
    add(placePanel, placePanel.toString());
    add(techPanel, techPanel.toString());
    add(endTurnPanel, endTurnPanel.toString());
    add(moveForumPosterPanel, moveForumPosterPanel.toString());
    add(politicsPanel, politicsPanel.toString());
    add(userActionPanel, userActionPanel.toString());
    add(pickTerritoryAndUnitsPanel, pickTerritoryAndUnitsPanel.toString()); 
```

In Container.java:
```
                } else if (constraints instanceof String) {
                    layoutMgr.addLayoutComponent((String)constraints, comp);
                }
```